sim/paladin/item_sets_pve.go

```diff
@@ -443,78 +204,482 @@ var ItemSetWilfullJudgement = core.NewItemSet(core.ItemSet{
 var ItemSetAvengersRadiance = core.NewItemSet(core.ItemSet{
 	Name: "Avenger's Radiance",
 	Bonuses: map[int32]core.ApplyEffect{
-		2: func(agent core.Agent) {
-			paladin := agent.(PaladinAgent).GetPaladin()
-			if !paladin.hasRune(proto.PaladinRune_RuneHandsCrusaderStrike) {
-				return
-			}
-
-			paladin.OnSpellRegistered(func(spell *core.Spell) {
-				//"S03 - Item - TAQ - Paladin - Retribution 2P Bonus",
-				if spell.SpellCode == SpellCode_PaladinCrusaderStrike {
-					// 2 Set: Increases Crusader Strike Damage by 50%
-					spell.DamageMultiplier += 0.5
-				}
-			})
-		},
-		4: func(agent core.Agent) {
-			paladin := agent.(PaladinAgent).GetPaladin()
-
-			buffAura := paladin.GetOrRegisterAura(core.Aura{
-				Label:     "Excommunication",
-				ActionID:  core.ActionID{SpellID: 1217927},
-				Duration:  time.Second * 20,
-				MaxStacks: 3,
-				OnStacksChange: func(aura *core.Aura, sim *core.Simulation, oldStacks int32, newStacks int32) {
-					for _, exorcism := range paladin.exorcism {
-						exorcism.DamageMultiplierAdditive += 0.4 * float64(newStacks-oldStacks)
-					}
-				},
-			})
-
-			core.MakePermanent(paladin.RegisterAura(core.Aura{
-				Label: "S03 - Item - TAQ - Paladin - Retribution 4P Bonus",
-				OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
-					if result.Landed() && spell.ProcMask.Matches(core.ProcMaskMeleeWhiteHit) {
-						buffAura.Activate(sim)
-						buffAura.AddStack(sim)
-					} else if spell.SpellCode == SpellCode_PaladinExorcism {
-						buffAura.Deactivate(sim)
-					}
-				},
-			}))
-		},
+		2: func(agent core.Agent) { agent.(PaladinAgent).GetPaladin().applyPaladinTAQRet2P() },
+		4: func(agent core.Agent) { agent.(PaladinAgent).GetPaladin().applyPaladinTAQRet4P() },
 	},
 })
 
 var ItemSetBattlegearOfEternalJustice = core.NewItemSet(core.ItemSet{
 	Name: "Battlegear of Eternal Justice",
 	Bonuses: map[int32]core.ApplyEffect{
 		// Crusader Strike now unleashes the judgement effect of your seals, but does not consume the seal
-		3: func(agent core.Agent) {
-			paladin := agent.(PaladinAgent).GetPaladin()
+		3: func(agent core.Agent) { agent.(PaladinAgent).GetPaladin().applyPaladinRAQ3P() },
+	},
+})
+
+func (paladin *Paladin) applyPaladinT1Prot2P() {
+	// (2) Set: Increases the block value of your shield by 30.
+	paladin.AddStat(stats.BlockValue, 30)
+}
```

Comment:
Won't this bonus stack if you wear the enchant + set bonus? You could apply it with an aura like this:
https://github.com/wowsims/sod/blob/4b20b13717962d5075f47df57f64df8ddf3cbb4d/sim/shaman/item_sets_pve_phase_4.go#L163-L192

Reply:
it would, but I went off of the assumption that stat stick auras would stack and unique effect auras would not. We still don't know exactly how it will work but I think it is a reasonable assumption.